### ISSUE TITLE
Way to create/buy custom video url

### ISSUE THREAD

**Issue by tibfox** [dtube/avalon#69]
It would be a great way to burn some tokens if we would sell custom video URLs within the uploader instead of the hash based URL. Especially for long-term content that would be nice to have

**Comment by techcoderx** [comment]
Of course, being an owner of a content on Avalon means they own the permlink of it, all metadata that belongs to it (JSON data, votes data etc) and its edit history which are currently not indexed by Avalon nodes.

**Comment by techcoderx** [comment]
We might have a problem if this is ever implemented

https://news.bitcoin.com/thailand-bans-meme-coins-fan-tokens-nfts-trading-crypto-exchanges/

**Comment by skzap** [comment]
We don't care about the fake news and regulations/bans attempts from people who have 0 actual control on avalon. Not a problem for me.

**Comment by brishtiteveja** [comment]
@tibfox I am thinking about getting a NFT tab on the mobile app after we make some progress on this one.
So, avalon needs to be extended for "Proof of content ownership"(NFT) and then "proof of transfer" I guess to enable selling NFTs.

@techcoderx You said "Avalon contents can be transferred to another account which enables the ability to transfer videos to another channel." So, let's say we define  
    hash("author", "ts", "content/comment hash") = Proof of ownership
  Is that enough?
  
  For NFT sell? 
     If a user wants to sell his content? 
       1. We create a market place (A separate page/tab) where new/trending NFTs will show up
       2. Owner(author) makes a new transaction (txtype: auctionContentAsNFT) that converts a comment into a NFT object
       3. We create a db.nft collection where we store the has of all NFTs (proof of ownership) mapped with its current owner, price auctioned, price bid, price sold etc. history for each nft
       4. A NFT buyer buys a nft by making a new transaction (txtype: buyContentAsNFT) which also triggers a new transaction (txtype: sellContentAsNFT) from the seller/owner's side 
       
  Any more ideas? Or more details @skzap ?

**Comment by techcoderx** [comment]
> hash("author", "ts", "content/comment hash") = Proof of ownership
Is that enough?

No, as the author may or may not be the owner. There will be a need for an `owner` property that is initially set to `author` on creation. When transferred/traded the `owner` property is set to the new owner but the `author` will always remain the same. The royalty fee (if applicable) goes to the author.

**Comment by brishtiteveja** [comment]
Yeah, that makes sense. I guess, I thought, currently author = owner, so may be when we add the new "owner" property, before we start NFT, we make owner = author by default, but after NFT, we will see many of the contents with owner not equals to author because of NFT sales.

**Comment by skzap** [comment]
Being able to transfer/sell videos to another account is an interesting idea and helpful to content creators.

Calling that a "NFT" feature is just marketing. Maybe would have been good in 2021, but doesn't seem as good in 2022.

**Comment by brishtiteveja** [comment]
@skzap essentially, it would be nice to create a marketplace like opensea where creators can exchange or sell to each other using DTC token.. and having a proof of ownership (NFT or whatever you wanna call..!!) is required.
